### PR TITLE
Add convertStrToVec and astring support in ParameterSet

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
To load a vector of things, use astring to load the whole parameter content.
Then use convertStrToVec to generate the vector.

## What type(s) of changes does this code introduce?
- New feature

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by ye-luo** [comment]
Test this please

File: src/Utilities/string_utils.h [hunk 1]
```diff
@@ -139,11 +142,30 @@ inline bool string2bool(const std::string& s)
...
   else
   {
-    APP_ABORT("string2bool received non-boolean string: " + s);
+    throw std::runtime_error("string2bool received non-boolean string: " + s);
     return false;
   }
 }
 
+template<class T>
+inline std::vector<T> convertStrToVec(const std::string& s)
```
Comment (markdewing) [inline src/Utilities/string_utils.h:151]: What format does this function expect the string to have?  Elements separated by spaces, commas, etc?

Comment (ye-luo) [reply src/Utilities/string_utils.h:151]: it is through `istringstream>>` so by white space. I should add a comment. Should I rename the function to `convertStrToVecBySpaces`?

Comment (markdewing) [reply src/Utilities/string_utils.h:151]: No, the separator shouldn't be in the name. It should be documented, though.

Comment (ye-luo) [reply src/Utilities/string_utils.h:151]: I will add a line of documentation in  #4168

**Review by PDoakORNL — APPROVED** [review]
This looks workable

**Pull request merged**